contrib/cf/pcf-tile/Makefile

```diff
@@ -0,0 +1,38 @@
+SHELL = /bin/bash
+
+GENERATE_TILE = docker run --rm -v ${CURDIR}:/workspace -w /workspace --name=tile-generator cfplatformeng/tile-generator:v11.0.5 tile build
+
+# Checks for the existence of a docker client and prints a nice error message
+# if it isn't present
+.PHONY: check-docker
+check-docker:
+	@if [ -z $$(which docker) ]; then \
+		echo "Missing \`docker\` client which is required for development"; \
+		exit 2; \
+	fi
+
+.PHONY: pull-tile-generator
+pull-tile-generator: check-docker
+	docker pull cfplatformeng/tile-generator:v11.0.5
+
+.PHONY: zip-broker
+zip-broker:
+	pushd ../../..; \
+	  rm -f contrib/cf/pcf-tile/resources/open-service-broker-azure.zip; \
+	  zip -r contrib/cf/pcf-tile/resources/open-service-broker-azure.zip cmd pkg vendor; \
+	popd;
+
+.PHONY: prepare-to-generate-tile
+prepare-to-generate-tile: pull-tile-generator zip-broker
+
+.PHONY: generate-major-tile
+generate-major-tile: prepare-to-generate-tile
```

Comment:
@zhongyi-zhang something I am struggling with here...

Since there are separate targets for generating major and minor versions of tiles, I need to know how we automate this. How can the CI/CD pipeline know which of these two targets to invoke?

We can could apply some regex to the version number, of course, but what I'm really trying to understand is why the process is _different_ for major vs minor. Should we even _need_ to decide between the two? How do the results of one differ from the other?

Reply:
From [docs](https://docs.pivotal.io/tiledev/2-0/tile-generator.html):

> Tile Generator uses semver versioning. By default, tile build generates the next patch release. Major and minor releases can be generated by explicitly specifying tile build major or tile build minor. Or to override the version number completely, specify a valid semver version on the build command, e.g. tile build 3.4.5.

So maybe we should have a single make target for generating tiles and it should learn the version number from an environment variable set by the CD pipeline?

Here's what we do for OSBA Docker images, for instance during CD (publishing to Docker Hub):

```
# Push officially released, semantically versioned image
.PHONY: push-release
push-release:
ifndef REL_VERSION
	$(error REL_VERSION is undefined)
endif
	docker pull $(RC_IMAGE_NAME)
	docker tag $(RC_IMAGE_NAME) $(REL_IMAGE_NAME)
	docker tag $(RC_IMAGE_NAME) $(REL_MUTABLE_IMAGE_NAME)
	docker push $(REL_IMAGE_NAME)
	docker push $(REL_MUTABLE_IMAGE_NAME)
```

I imagine, perhaps, we can do something similar for the tile:

```
.PHONY: generate-tile
generate-tile: prepare-to-generate-tile
@# REL_VERSION will be set by the CD pipeline when publishing a tile (i.e. as part of the release process)
@# When REL_VERSION is not set, we'll generate without specifying a version number. This simply allows
@# this step to be tested during CI so that we know it will work in CD.
ifdef REL_VERSION
	${GENERATE_TILE} ${REL_VERSION}
else
	${GENERATE_TILE}
endif
```

Then specific targets `generate-major-tile` and `generate-minor-tile` go away.

Let me know if you think this works.